demos/stokes/setup.py

```diff
@@ -1,7 +1,5 @@
+from animate.metric import RiemannianMetric
```

Comment:
(not commenting at the relevant line)

Uniform and hessian raise
```
(firedrake) firedrake@e4a48468346d:~/firedrake/src/opt_adapt/demos$ python opt_hessian.py 'stokes'
/home/firedrake/firedrake/lib/python3.12/site-packages/pytools/persistent_dict.py:52: RecommendedHashNotFoundWarning: Unable to import recommended hash 'siphash24.siphash13', falling back to 'hashlib.sha256'. Run 'python3 -m pip install siphash24' to install the recommended hash.
  warn("Unable to import recommended hash 'siphash24.siphash13', "
Using method gradient_descent
  1:  J = 4.6078e+01,  ||dJdm|| = 1.1935e+02,  step length = 1.0000e-03,  #elements =   746,  time = 0.42s
Traceback (most recent call last):
  File "/home/firedrake/firedrake/src/opt_adapt/demos/opt_hessian.py", line 112, in <module>
    m_opt = minimise(
            ^^^^^^^^^
  File "/home/firedrake/firedrake/src/opt_adapt/opt_adapt/opt.py", line 555, in minimise
    for o in step(it, forward_run, mesh, params, *args):
  File "/home/firedrake/firedrake/src/opt_adapt/opt_adapt/opt.py", line 222, in _gradient_descent
    u += lr * P
  File "/home/firedrake/firedrake/src/firedrake/firedrake/adjoint_utils/function.py", line 146, in wrapper
    func = __iadd__(self, other, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake/src/firedrake/firedrake/function.py", line 519, in __iadd__
    IAddAssigner(self, expr).assign()
    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake/src/firedrake/firedrake/assign.py", line 147, in __init__
    expression = as_ufl(expression)
                 ^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake/src/ufl/ufl/constantvalue.py", line 517, in as_ufl
    raise ValueError(
ValueError: Invalid type conversion: [Product(FloatValue(7.355163906238338e-05), Indexed(Coefficient(WithGeometry(FunctionSpace(<firedrake.mesh.MeshTopology object at 0x7f043179b5c0>, VectorElement(FiniteElement('Lagrange', triangle, 2), dim=2), name=None), Mesh(VectorElement(FiniteElement('Lagrange', triangle, 1), dim=2), 117)), 323), MultiIndex((FixedIndex(0),))))
 Product(FloatValue(7.355163906238338e-05), Indexed(Coefficient(WithGeometry(FunctionSpace(<firedrake.mesh.MeshTopology object at 0x7f043179b5c0>, VectorElement(FiniteElement('Lagrange', triangle, 2), dim=2), name=None), Mesh(VectorElement(FiniteElement('Lagrange', triangle, 1), dim=2), 117)), 323), MultiIndex((FixedIndex(1),))))] can not be converted to any UFL type.
 ```
 
 and GO raises
 ```
 (firedrake) firedrake@e4a48468346d:~/firedrake/src/opt_adapt/demos$ python opt_go.py 'stokes'
/home/firedrake/firedrake/lib/python3.12/site-packages/pytools/persistent_dict.py:52: RecommendedHashNotFoundWarning: Unable to import recommended hash 'siphash24.siphash13', falling back to 'hashlib.sha256'. Run 'python3 -m pip install siphash24' to install the recommended hash.
  warn("Unable to import recommended hash 'siphash24.siphash13', "
Using method gradient_descent
  1:  J = 4.6078e+01,  ||dJdm|| = 1.1935e+02,  step length = 1.0000e-03,  #elements =   746,  time = 0.45s
Traceback (most recent call last):
  File "/home/firedrake/firedrake/src/opt_adapt/demos/opt_go.py", line 164, in <module>
    m_opt = minimise(
            ^^^^^^^^^
  File "/home/firedrake/firedrake/src/opt_adapt/opt_adapt/opt.py", line 555, in minimise
    for o in step(it, forward_run, mesh, params, *args):
  File "/home/firedrake/firedrake/src/opt_adapt/opt_adapt/opt.py", line 202, in _gradient_descent
    J, u = forward_run(m, u, **params.model_options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake/src/opt_adapt/demos/stokes/setup.py", line 68, in forward_run
    outfile.write(u, p)
  File "petsc4py/PETSc/Log.pyx", line 188, in petsc4py.PETSc.Log.EventDecorator.decorator.wrapped_func
  File "petsc4py/PETSc/Log.pyx", line 189, in petsc4py.PETSc.Log.EventDecorator.decorator.wrapped_func
  File "/home/firedrake/firedrake/src/firedrake/firedrake/output/vtk_output.py", line 653, in write
    vtu = self._write_vtu(*functions)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake/src/firedrake/firedrake/output/vtk_output.py", line 519, in _write_vtu
    raise ValueError("Writing different set of functions")
ValueError: Writing different set of functions
```

Reply:
Related to #30.